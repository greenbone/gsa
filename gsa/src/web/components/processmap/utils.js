/* Copyright (C) 2020 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';

import styled from 'styled-components';

import {_} from 'gmp/locale/lang';

import PropTypes from 'web/utils/proptypes';

export const createTag = ({name, gmp}) => {
  return gmp.tag
    .create({
      active: '1',
      name,
      resource_type: 'host',
    })
    .then(response => {
      const {data = {}} = response;
      return data.id;
    });
};

export const editTag = ({action = 'add', name, hostIds, tagId, gmp}) => {
  return gmp.tag.save({
    active: '1',
    name,
    id: tagId,
    resource_ids: hostIds,
    resource_type: 'host',
    resources_action: action, // this always needs to be sent, even without
    // adding new resources in order to prevent
    // emptying the list when only changing the name
  });
};

export const deleteTag = ({tagId, gmp}) => {
  return gmp.tag.delete({
    id: tagId,
  });
};

const ToolTipContent = styled.div`
  font-weight: normal;
  text-align: center;
  line-height: 1.2em;
`;

export const createToolTipContent = ({severity}) => {
  return (
    <ToolTipContent>
      {severity < 0 ? (
        _('No original severity value available')
      ) : (
        <span>
          {_('Original severity: ')}
          <b>{severity}</b>
        </span>
      )}
    </ToolTipContent>
  );
};

createToolTipContent.propTypes = {
  severity: PropTypes.number,
};

// vim: set ts=2 sw=2 tw=80:
