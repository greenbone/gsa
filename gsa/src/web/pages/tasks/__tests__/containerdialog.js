/* Copyright (C) 2019 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import React from 'react';

import {render, fireEvent} from 'web/utils/testing';

import ContainerDialog from '../containerdialog';
import Task, {
  AUTO_DELETE_KEEP,
  AUTO_DELETE_KEEP_DEFAULT_VALUE,
  AUTO_DELETE_NO,
} from 'gmp/models/task';

describe('ContainerDialog tests', () => {
  test('should render create dialog', () => {
    const handleClose = jest.fn();
    const handleSave = jest.fn();

    const {baseElement} = render(
      <ContainerDialog onClose={handleClose} onSave={handleSave} />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('should render edit dialog', () => {
    const task = Task.fromElement({name: 'foo', _id: 't1'});
    const handleClose = jest.fn();
    const handleSave = jest.fn();

    const {baseElement} = render(
      <ContainerDialog task={task} onClose={handleClose} onSave={handleSave} />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('should change fields in create dialog', () => {
    const handleClose = jest.fn();
    const handleSave = jest.fn();

    const {getByName, getByTestId} = render(
      <ContainerDialog
        name="foo"
        comment="bar"
        onClose={handleClose}
        onSave={handleSave}
      />,
    );

    const nameInput = getByName('name');
    fireEvent.change(nameInput, {target: {value: 'ipsum'}});

    const commentInput = getByName('comment');
    fireEvent.change(commentInput, {target: {value: 'lorem'}});

    const saveButton = getByTestId('dialog-save-button');
    fireEvent.click(saveButton);

    expect(handleSave).toHaveBeenCalledWith({
      comment: 'lorem',
      auto_delete: AUTO_DELETE_KEEP,
      auto_delete_data: AUTO_DELETE_KEEP_DEFAULT_VALUE,
      in_assets: 1,
      id: undefined,
      name: 'ipsum',
    });
  });

  test('should change fields in edit dialog', () => {
    const task = Task.fromElement({name: 'foo', _id: 't1'});
    const handleClose = jest.fn();
    const handleSave = jest.fn();

    const {getByName, queryAllByName, getByTestId} = render(
      <ContainerDialog
        task={task}
        name="foo"
        comment="bar"
        onClose={handleClose}
        onSave={handleSave}
      />,
    );

    const nameInput = getByName('name');
    fireEvent.change(nameInput, {target: {value: 'ipsum'}});

    const commentInput = getByName('comment');
    fireEvent.change(commentInput, {target: {value: 'lorem'}});

    const autoDeleteKeepInput = getByName('auto_delete_data');
    fireEvent.change(autoDeleteKeepInput, {target: {value: '10'}});

    const [autoDeleteNoRadio] = queryAllByName('auto_delete');
    fireEvent.click(autoDeleteNoRadio);

    const [, inAssetsNoRadio] = queryAllByName('in_assets');
    fireEvent.click(inAssetsNoRadio);

    const saveButton = getByTestId('dialog-save-button');
    fireEvent.click(saveButton);

    expect(handleSave).toHaveBeenCalledWith({
      comment: 'lorem',
      auto_delete: AUTO_DELETE_NO,
      auto_delete_data: 10,
      in_assets: 0,
      id: 't1',
      name: 'ipsum',
    });
  });

  test('should allow to close the dialog', () => {
    const handleClose = jest.fn();
    const handleSave = jest.fn();

    const {getByTestId} = render(
      <ContainerDialog
        name="foo"
        comment="bar"
        onClose={handleClose}
        onSave={handleSave}
      />,
    );

    const closeButton = getByTestId('dialog-close-button');

    fireEvent.click(closeButton);

    expect(handleClose).toHaveBeenCalled();
    expect(handleSave).not.toHaveBeenCalled();
  });
});
