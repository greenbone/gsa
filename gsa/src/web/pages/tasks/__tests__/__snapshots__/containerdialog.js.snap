// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerDialog tests should render create dialog 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10% auto;
  border: 0;
  outline: 0;
  width: 800px;
  height: px;
}

.c0 {
  position: fixed;
  font-family: Trebuchet MS,Tahoma,Verdana,Arial,sans-serif;
  font-size: 1.1em;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: 0;
  background: rgba(102,102,102,0.5);
  z-index: 600;
  -webkit-transition: opacity 1s ease-in;
  transition: opacity 1s ease-in;
  width: 100%;
  height: 100%;
}

.c19 {
  width: 0;
  height: 0;
  cursor: nwse-resize;
  border-right: 20px solid transparent;
  border-bottom: 20px solid transparent;
  border-top: 20px solid #fff;
}

.c18 {
  position: absolute;
  bottom: 3px;
  right: 3px;
  width: 20px;
  height: 20px;
  background: repeating-linear-gradient( -45deg, transparent, transparent 2px, #000 2px, #000 3px );
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: inherit;
  padding: 0;
  background: #fff;
  box-shadow: 5px 5px 10px #787878;
  border-radius: 3px;
  border: 1px solid #787878;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #519032;
  font-weight: bold;
  font-size: 12px;
  font-family: Verdana,sans-serif;
  color: #519032;
  cursor: pointer;
  border-radius: 2px;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4:hover {
  color: #fff;
  background: #519032;
}

.c5 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c5 * {
  height: inherit;
  width: inherit;
}

.c15 {
  display: inline-block;
  padding: 0 15px;
  color: #393637;
  text-align: center;
  vertical-align: middle;
  font-size: 11px;
  font-weight: bold;
  line-height: 30px;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  background-color: #fff;
  border-radius: 2px;
  border: 1px solid #aaaaaa;
  cursor: pointer;
  overflow: visible;
  z-index: 1;
}

.c15:focus,
.c15:hover {
  border: 1px solid #393637;
}

.c15:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #66c430;
  font-weight: bold;
  color: #fff;
}

.c15[disabled] {
  cursor: not-allowed;
  opacity: 0.65;
  box-shadow: none;
}

.c15 img {
  height: 32px;
  width: 32px;
  margin-top: 5px 10px 5px -10px;
  vertical-align: middle;
}

.c15:link {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #393637;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 {
  border: 1px solid #787878;
  color: #519032;
  background: #87d050;
}

.c17:hover {
  color: #fff;
  background: #519032;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-width: 1px 0 0 0;
  border-style: solid;
  border-color: #c8d3d9;
  margin-top: 15px;
  padding: 10px 20px 10px 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 5px 5px 10px;
  margin-bottom: 15px;
  border-radius: 2px 2px 0 0;
  border-bottom: 1px solid #787878;
  color: #fff;
  font-weight: bold;
  background: #66c430;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c7 {
  overflow: auto;
  padding: 0 15px;
  width: 100%;
  height: 100%;
  max-height: 400px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  height: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding-bottom: 10px;
}

.c10 {
  display: inline-block;
  max-width: 100%;
  font-weight: bold;
  text-align: right;
  padding-left: 10px;
  padding-right: 10px;
  width: 16.66666667%;
  margin-left: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 83.33333333%;
  padding-left: 10px;
  padding-right: 10px;
}

.c12 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  display: block;
  height: 22px;
  color: #393637;
  background-color: #fff;
  background-image: none;
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  padding: 1px 8px;
}

.c12:-webkit-autofill {
  box-shadow: 0 0 0 1000px white inset;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<body>
  <div
    id="portals"
  >
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          height="px"
          role="dialog"
          tabindex="1"
          width="800px"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
              data-testid="dialog-title-bar"
            >
              <span>
                New Container Task
              </span>
              <div
                class="c4 c5"
                data-testid="dialog-title-close-button"
                title="Close"
              >
                Ã—
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c7"
                data-testid="save-dialog-content"
              >
                <div
                  class="c8"
                >
                  <div
                    class="c9"
                  >
                    <label
                      class="c10"
                      data-testid="formgroup-title"
                    >
                      Name
                    </label>
                    <div
                      class="c11"
                      data-testid="formgroup-content"
                      size="10"
                    >
                      <input
                        class="c12 c13"
                        name="name"
                        size="30"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c9"
                  >
                    <label
                      class="c10"
                      data-testid="formgroup-title"
                    >
                      Comment
                    </label>
                    <div
                      class="c11"
                      data-testid="formgroup-content"
                      size="10"
                    >
                      <input
                        class="c12 c13"
                        name="comment"
                        size="30"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c14"
            >
              <button
                class="c15 c16 c17"
                data-testid="dialog-close-button"
                title="Cancel"
              >
                Cancel
              </button>
              <button
                class="c15 c16 c17"
                data-testid="dialog-save-button"
                title="Save"
              >
                Save
              </button>
            </div>
          </div>
          <div
            class="c18"
          >
            <div
              class="c19"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</body>
`;

exports[`ContainerDialog tests should render edit dialog 1`] = `
<body>
  <div
    id="portals"
  >
    .c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c16 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c16 > * {
  margin-left: 5px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10% auto;
  border: 0;
  outline: 0;
  width: 800px;
  height: px;
}

.c0 {
  position: fixed;
  font-family: Trebuchet MS,Tahoma,Verdana,Arial,sans-serif;
  font-size: 1.1em;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: 0;
  background: rgba(102,102,102,0.5);
  z-index: 600;
  -webkit-transition: opacity 1s ease-in;
  transition: opacity 1s ease-in;
  width: 100%;
  height: 100%;
}

.c26 {
  width: 0;
  height: 0;
  cursor: nwse-resize;
  border-right: 20px solid transparent;
  border-bottom: 20px solid transparent;
  border-top: 20px solid #fff;
}

.c25 {
  position: absolute;
  bottom: 3px;
  right: 3px;
  width: 20px;
  height: 20px;
  background: repeating-linear-gradient( -45deg, transparent, transparent 2px, #000 2px, #000 3px );
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: inherit;
  padding: 0;
  background: #fff;
  box-shadow: 5px 5px 10px #787878;
  border-radius: 3px;
  border: 1px solid #787878;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #519032;
  font-weight: bold;
  font-size: 12px;
  font-family: Verdana,sans-serif;
  color: #519032;
  cursor: pointer;
  border-radius: 2px;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4:hover {
  color: #fff;
  background: #519032;
}

.c5 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c5 * {
  height: inherit;
  width: inherit;
}

.c22 {
  display: inline-block;
  padding: 0 15px;
  color: #393637;
  text-align: center;
  vertical-align: middle;
  font-size: 11px;
  font-weight: bold;
  line-height: 30px;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  background-color: #fff;
  border-radius: 2px;
  border: 1px solid #aaaaaa;
  cursor: pointer;
  overflow: visible;
  z-index: 1;
}

.c22:focus,
.c22:hover {
  border: 1px solid #393637;
}

.c22:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #66c430;
  font-weight: bold;
  color: #fff;
}

.c22[disabled] {
  cursor: not-allowed;
  opacity: 0.65;
  box-shadow: none;
}

.c22 img {
  height: 32px;
  width: 32px;
  margin-top: 5px 10px 5px -10px;
  vertical-align: middle;
}

.c22:link {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #393637;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c24 {
  border: 1px solid #787878;
  color: #519032;
  background: #87d050;
}

.c24:hover {
  color: #fff;
  background: #519032;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-width: 1px 0 0 0;
  border-style: solid;
  border-color: #c8d3d9;
  margin-top: 15px;
  padding: 10px 20px 10px 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 5px 5px 10px;
  margin-bottom: 15px;
  border-radius: 2px 2px 0 0;
  border-bottom: 1px solid #787878;
  color: #fff;
  font-weight: bold;
  background: #66c430;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
}

.c7 {
  overflow: auto;
  padding: 0 15px;
  width: 100%;
  height: 100%;
  max-height: 400px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  height: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding-bottom: 10px;
}

.c10 {
  display: inline-block;
  max-width: 100%;
  font-weight: bold;
  text-align: right;
  padding-left: 10px;
  padding-right: 10px;
  width: 16.66666667%;
  margin-left: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 83.33333333%;
  padding-left: 10px;
  padding-right: 10px;
}

.c12 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  display: block;
  height: 22px;
  color: #393637;
  background-color: #fff;
  background-image: none;
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  padding: 1px 8px;
}

.c12:-webkit-autofill {
  box-shadow: 0 0 0 1000px white inset;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: normal;
  cursor: pointer;
}

.c18 {
  font-family: inherit;
  font-size: inherit;
  padding: 0;
  margin: 0;
  margin-left: 10px;
  line-height: normal;
  width: auto;
  height: auto;
}

.c20 {
  opacity: 1;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
          height="px"
          role="dialog"
          tabindex="1"
          width="800px"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
              data-testid="dialog-title-bar"
            >
              <span>
                New Container Task
              </span>
              <div
                class="c4 c5"
                data-testid="dialog-title-close-button"
                title="Close"
              >
                Ã—
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c7"
                data-testid="save-dialog-content"
              >
                <div
                  class="c8"
                >
                  <div
                    class="c9"
                  >
                    <label
                      class="c10"
                      data-testid="formgroup-title"
                    >
                      Name
                    </label>
                    <div
                      class="c11"
                      data-testid="formgroup-content"
                      size="10"
                    >
                      <input
                        class="c12 c13"
                        name="name"
                        size="30"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c9"
                  >
                    <label
                      class="c10"
                      data-testid="formgroup-title"
                    >
                      Comment
                    </label>
                    <div
                      class="c11"
                      data-testid="formgroup-content"
                      size="10"
                    >
                      <input
                        class="c12 c13"
                        name="comment"
                        size="30"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c9"
                  >
                    <label
                      class="c10"
                      data-testid="formgroup-title"
                    >
                      Add results to Assets
                    </label>
                    <div
                      class="c11"
                      data-testid="formgroup-content"
                      size="10"
                    >
                      <div
                        class="c14"
                      >
                        <div
                          class="c15"
                        >
                          <div
                            class="c16"
                            margin="5px"
                          >
                            <label
                              class="c17"
                            >
                              <div
                                class="c15"
                              >
                                <div
                                  class="c16"
                                  margin="5px"
                                >
                                  <input
                                    checked=""
                                    class="c18 c19"
                                    data-testid="radio-input"
                                    name="in_assets"
                                    type="radio"
                                    value="1"
                                  />
                                  <span
                                    class="c20"
                                    data-testid="radio-title"
                                  >
                                    Yes
                                  </span>
                                </div>
                              </div>
                            </label>
                            <label
                              class="c17"
                            >
                              <div
                                class="c15"
                              >
                                <div
                                  class="c16"
                                  margin="5px"
                                >
                                  <input
                                    class="c18 c19"
                                    data-testid="radio-input"
                                    name="in_assets"
                                    type="radio"
                                    value="0"
                                  />
                                  <span
                                    class="c20"
                                    data-testid="radio-title"
                                  >
                                    No
                                  </span>
                                </div>
                              </div>
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c21"
            >
              <button
                class="c22 c23 c24"
                data-testid="dialog-close-button"
                title="Cancel"
              >
                Cancel
              </button>
              <button
                class="c22 c23 c24"
                data-testid="dialog-save-button"
                title="Save"
              >
                Save
              </button>
            </div>
          </div>
          <div
            class="c25"
          >
            <div
              class="c26"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</body>
`;
